apps/src/javalab/JavalabDialog.jsx

```diff
@@ -45,16 +50,24 @@ export default class JavalabDialog extends Component {
         >
           <div style={styles.message}>{message}</div>
           <div style={styles.buttons}>
+            {showSpinner && (
+              <i
+                className={classNames('fa', 'fa-spin', 'fa-spinner')}
+                style={styles.spinner}
+              />
+            )}
             {closeButtonText && (
               <button
                 type="button"
                 style={{
                   ...styles.button,
                   ...(displayTheme === DisplayTheme.DARK
                     ? styles.darkButton
-                    : styles.lightCancel)
+                    : styles.lightCancel),
+                  ...(disableButtons ? styles.disabledButton : {})
```

Comment:
Where is `styles.disabledButton` defined? Didn't see it in the style object so just wanted to confirm

Reply:
Good catch. I initially was going to gray out the buttons but decided against it because the cancel button is already grey and in dark mode the buttons are both black. Since the buttons are only disabled to prevent double clicks and the delete should go pretty quick I decided to just have them look the same but not be clickable while delete is in progress. There is a spinner to indicate something is happening if it does go slowly.